packages/engine/src/scene/materials/systems/MaterialLibrarySystem.tsx

```diff
@@ -89,7 +89,7 @@ const MaterialEntityReactor = () => {
   const entity = useEntityContext()
   const materialComponent = useComponent(entity, MaterialStateComponent)
   useEffect(() => {
-    if (!materialComponent.instances.value! || !SourceComponent.entitiesBySource) return
+    if (!materialComponent.instances.value! || !Object.keys(SourceComponent.entitiesBySource)) return
```

Comment:
by checking for object keys here, we escaping all conditionals rather than checking if sourceEntity has a source component and if entitiesBySource has data for that entity? that seems to be a more appropriate check

Reply:
Yes you're right - my bad for multitasking while debugging this